analyzers/tests/SonarAnalyzer.TestFramework/Verification/IssueValidation/CompilationIssues.cs

```diff
@@ -18,51 +18,70 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
+using System.Collections;
 using System.IO;
 
 namespace SonarAnalyzer.TestFramework.Verification.IssueValidation;
 
-internal sealed class CompilationIssues
+internal sealed class CompilationIssues : IEnumerable<IssueLocation>
 {
-    public string LanguageVersion { get; }
-    private readonly FileIssueLocations[] fileIssues;
+    public string LanguageVersion { get; }  // ToDo: Should this be here?
+    private readonly List<IssueLocation> issues;
 
     public CompilationIssues(string languageVersion, IEnumerable<FileContent> files)
+        : this(languageVersion, files.SelectMany(x => IssueLocationCollector.GetExpectedIssueLocations(x.Content.Lines))) { }
+
+    public CompilationIssues(string languageVersion, Diagnostic[] diagnostics)
+        : this(languageVersion, ToIssueLocations(diagnostics)) { }
+
+    public CompilationIssues(string languageVersion, IEnumerable<IssueLocation> issues)
     {
         LanguageVersion = languageVersion;
-        fileIssues = files.Select(x => new FileIssueLocations(x.FileName, IssueLocationCollector.GetExpectedIssueLocations(x.Content.Lines))).ToArray();
+        this.issues = issues.ToList();
     }
 
-    public CompilationIssues(string languageVersion, Diagnostic[] diagnostics)
+    public IEnumerable<IssueLocationKey> UniqueKeys() =>
+        issues.Select(x => new IssueLocationKey(x)).Distinct();
+
+    public List<IssueLocation> Remove(IssueLocationKey key)
     {
-        var map = new Dictionary<string, List<IssueLocation>>();
-        foreach (var diagnostic in diagnostics)
+        var ret = issues.Where(key.IsMatch).ToList();
+        foreach (var issue in ret)
         {
-            Add(new IssueLocation(diagnostic));
-            for (var i = 0; i < diagnostic.AdditionalLocations.Count; i++)
-            {
-                Add(new IssueLocation(diagnostic.GetSecondaryLocation(i)));
-            }
+            issues.Remove(issue);
         }
-        LanguageVersion = languageVersion;
-        fileIssues = map.Select(x => new FileIssueLocations(x.Key, x.Value)).ToArray();
+        return ret;
+    }
 
-        void Add(IssueLocation issue)
+    public void Dump()
+    {
+        foreach (var file in issues.GroupBy(x => Path.GetFileName(x.FilePath)).OrderBy(x => x.Key))
         {
-            var list = map.GetOrAdd(issue.FilePath ?? string.Empty, _ => new List<IssueLocation>());
-            list.Add(issue);
+            Console.WriteLine($"Actual {LanguageVersion} diagnostics {file.Key}:");
+            foreach (var issue in file.OrderBy(x => x.LineNumber))
+            {
+                Console.WriteLine($"    {issue.RuleId}, Line: {issue.LineNumber}, [{issue.Start}, {issue.Length}] {issue.Message}");
```

Comment:
Isn't there some `tab` symbol to add?
for example "Console.WriteLine($"\t{issue.RuleId}," instead of putting it manually?
Not strong against it but it is a bit confusing - I thought it was there by mistake.

Reply:
I don't want to risk it.. it might bite if we copy the content one day